### ISSUE TITLE
Add chromatic aberration correction

### ISSUE THREAD

**Issue by Consti10** [googlevr/cardboard#5]
Any plans on supporting chromatic aberration distortion correction with cardboard ?

Or at least open sourcing the daydream parameters ?

**Comment by jballoffet** [comment]
Thank you for submitting your feature request. We do not have short term plans to support chromatic aberration correction in the Cardboard SDK yet because we’re busily working on some other features. We’ll put it on the backlog. It’s also worth noting that chromatic aberration correction varies by lens design and can be expensive to perform. However, if you do want to try it out in the meantime because you have a known lens design and device with ample GPU performance, consider taking a look at [this function](https://github.com/googlevr/cardboard/blob/master/sdk/lens_distortion.cc#L89), which computes the geometric distortion for a single point. Note that the SDK implements a mesh approach to generate the distortion mesh, in order to avoid high latencies due to massive single point calculations. Taking that in mind, you could follow a similar approach to correct the chromatic aberration. Keep in mind that the distortion inverts the optical distortion caused by the lens for the eye and in this case, due to chromatic aberration, the distortion should be different for each color channel.

**Comment by Consti10** [comment]
Hello jballoffet, thanks for pointing that out.
I assume for chromatic aberration we only need

- [ ]  k1...kN radial distortion factors for the different color channels. 

- [ ]  different r,g,b values for the distortion_mesh.

I'm currently busy to add cardboard to my vertex displacement distortion correction implementation [https://github.com/googlevr/cardboard/issues/10](url) but I'm also planning to create a distortion renderer for shapes other than rectangles (e.g. 360° video sphere) with correction for chromatic aberration. That's still missing in gvr, by the way :)

**Comment by jballoffet** [comment]
Yes, as you said, you need to calculate the distortion mesh for each color channel, using the proper distortion factors for one of each. Let us know how that goes!

**Comment by mwsunshine** [comment]
> Thank you for submitting your feature request. We do not have short term plans to support chromatic aberration correction in the Cardboard SDK yet because we’re busily working on some other features. We’ll put it on the backlog. It’s also worth noting that chromatic aberration correction varies by lens design and can be expensive to perform. However, if you do want to try it out in the meantime because you have a known lens design and device with ample GPU performance, consider taking a look at [this function](https://github.com/googlevr/cardboard/blob/master/sdk/lens_distortion.cc#L89), which computes the geometric distortion for a single point. Note that the SDK implements a mesh approach to generate the distortion mesh, in order to avoid high latencies due to massive single point calculations. Taking that in mind, you could follow a similar approach to correct the chromatic aberration. Keep in mind that the distortion inverts the optical distortion caused by the lens for the eye and in this case, due to chromatic aberration, the distortion should be different for each color channel.

hi，is there any progress about the chromatic aberration correction made or any supporting plans in the short term as two years passed by?

**Comment by jballoffet** [comment]
This feature has not been prioritized in the short run, nevertheless, if you want to send your proposed changes, please read CONTRIBUTING.md and follow the process. Patches are welcome. Thanks!
